bindings/ios/MEGABackgroundMediaUpload.mm

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file MEGABackgroundMediaUpload.mm
+ * @brief Time zone details
+ *
+ * (c) 2018 - by Mega Limited, Auckland, New Zealand
+ *
+ * This file is part of the MEGA SDK - Client Access Engine.
+ *
+ * Applications using the MEGA API must present a valid application key
+ * and comply with the the rules set forth in the Terms of Service.
+ *
+ * The MEGA SDK is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * @copyright Simplified (2-clause) BSD License.
+ *
+ * You should have received a copy of the license along with this
+ * program.
+ */
+
+#import "MEGABackgroundMediaUpload.h"
+#import "megaapi.h"
+#import "MEGABackgroundMediaUpload+init.h"
+
+@interface MEGABackgroundMediaUpload ()
+
+@property (nonatomic) mega::MegaBackgroundMediaUpload *mediaUpload;
+
+@end
+
+@implementation MEGABackgroundMediaUpload
+
+- (instancetype)initWithBackgroundMediaUpload:(mega::MegaBackgroundMediaUpload *)mediaUpload {
+    self = [super init];
+    if (self) {
+        _mediaUpload = mediaUpload;
+    }
+    return self;
+}
+
+- (void)dealloc {
+    delete _mediaUpload;
+}
+
+- (mega::MegaBackgroundMediaUpload *)getCPtr {
+    return self.mediaUpload;
+}
+
+- (BOOL)encryptFileAtPath:(NSString *)inputFilePath startPosition:(int64_t)start length:(unsigned *)length outputFilePath:(NSString *)outputFilePath urlSuffix:(NSString **)urlSuffix adjustsSizeOnly:(BOOL)adjustsSizeOnly {
+    std::string suffix;
+    BOOL succeed = self.mediaUpload->encryptFile(inputFilePath.UTF8String, start, length, outputFilePath.UTF8String, &suffix, adjustsSizeOnly);
```

Comment:
Usually we pass to the intermediate layer an UTF8 string of NULL for const char * arguments, depending if the NSString is `nil` or not. I am not sure if we can/must do the same here, it depends how it is implemented in the c++ intermediate layer.

Reply:
I think technically, nil and NULL are same and equal to 0. And usually where you can use NULL you can pass nil. Here is the define of nil in MacTypes.h

> #ifndef nil
  #if defined(__has_feature) 
    #if __has_feature(cxx_nullptr)
      #define nil nullptr
    #else
      #define nil __DARWIN_NULL
    #endif
  #else
    #define nil __DARWIN_NULL
  #endif
#endif

However, for consistent and make the code easier to understand, I am happy to convert nil to NULL explicitly

Reply:
Yes, they are same. Thanks Navarro!